### PR TITLE
feat: Support multiple update types

### PR DESCRIPTION
**Pull request by zharinov**
Ref #6791

### CODE REVIEW

File: lib/config/definitions.ts [hunk 1]
```diff
@@ -883,7 +883,7 @@ const options: RenovateOptions[] = [
       'bump',
     ],
     allowString: true,
-    stage: 'package',
+    stage: 'branch',
```
Comment (zharinov) [inline lib/config/definitions.ts:886]: It fixes name conflict, but I guess it's potentially dangerous

File: lib/workers/repository/updates/flatten.ts [hunk 2]
```diff
@@ -51,10 +51,12 @@ export async function flattenUpdates(
...
-            updateConfig = mergeChildConfig(
-              updateConfig,
-              updateConfig[updateConfig.updateType]
-            );
+            updateConfig.updateTypes?.forEach((updateType) => {
+              updateConfig = mergeChildConfig(
+                updateConfig,
+                updateConfig[updateType]
+              );
+            });
```
Comment (zharinov) [inline lib/workers/repository/updates/flatten.ts:59]: Another one potentially dangerous place, though it's working for now (with single values)

File: lib/config/definitions.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/definitions.ts:886]: We plan to rename the package file one to be matchUpdateTypes

File: lib/workers/repository/process/sort.ts [hunk 3]
```diff
@@ -16,8 +16,10 @@ export function sortBranches(branches: Partial<BranchConfig>[]): void {
     if (a.prPriority !== b.prPriority) {
       return b.prPriority - a.prPriority;
     }
+    const aUpdateType = a.updateTypes ? a.updateTypes[0] : null;
+    const bUpdateType = b.updateTypes ? b.updateTypes[0] : null;
     const sortDiff =
-      sortOrder.indexOf(a.updateType) - sortOrder.indexOf(b.updateType);
+      sortOrder.indexOf(aUpdateType) - sortOrder.indexOf(bUpdateType);
```
Comment (viceice) [inline lib/workers/repository/process/sort.ts:22]: So this now depends on order of assigned update types? We need to think about this sorting

**Comment by rarkins** [comment]
@zharinov let's implement #6805 first

File: lib/config/common.ts [hunk 4]
```diff
@@ -165,7 +165,7 @@ export interface RenovateConfig
   prHourlyLimit?: number;
   repoIsOnboarded?: boolean;
 
-  updateType?: UpdateType;
+  matchUpdateTypes?: UpdateType[];
```
Comment (rarkins) [inline lib/config/common.ts:168]: Sorry, we maybe had a confusion here. The `updateTypes` in `packageRules` should be renamed to `matchUpateTypes`. The `updateType` that's returned from our lookup routine should be renamed `updateTypes`.

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/common.ts (hunk 4 above)
Comment (zharinov) [reply lib/config/common.ts:168]: Ah, sorry 🙃

**Comment by zharinov** [comment]
- [ ] Do we still need it?
- [ ] If so, start it over

**Comment by rarkins** [comment]
I'm not certain about this one. Let's park it in Draft for now.

**Pull request closed without merging**